helpers/chatHelper.py

```diff
@@ -266,462 +264,629 @@ async def partChannel(
+                "recipient_user_id": recipient_token["user_id"],
+            },
+        )
+
+        return ...
+
+    if osuToken.is_restricted(recipient_token["privileges"]):
+        logger.warning(
+            "User attempted to send a message to a restricted user",
+            extra={
+                "token_id": user_token["token_id"],
+                "username": user_token["username"],
+                "user_id": user_token["user_id"],
+                "recipient_username": recipient_token["username"],
+                "recipient_user_id": recipient_token["user_id"],
+            },
+        )
+        return ...
+
+    # TODO: check if user has us blacklisted
+    # fmt: off
+    if (
+        recipient_token["block_non_friends_dm"]
+        and user_token["user_id"]
+        not in await userUtils.getFriendList(recipient_token["user_id"])
+    ):
+    # fmt: on
+        await osuToken.enqueue(
+            user_token["token_id"],
+            serverPackets.targetBlockingDMs(recipient_token["username"]),
+        )
+
         logger.warning(
             "User tried to send a message to a user that is blocking non-friends dms",
             extra={
-                "username": userToken["username"],
-                "user_id": userToken["user_id"],
-                "recipient": to,
+                "token_id": user_token["token_id"],
+                "username": user_token["username"],
+                "user_id": user_token["user_id"],
+                "recipient_username": recipient_token["username"],
+                "recipient_user_id": recipient_token["user_id"],
+            },
+        )
+
+        return ...
+
+    if await osuToken.awayCheck(
+        recipient_token["token_id"],
+        user_token["user_id"],
+    ):
+        await enqueue_private_message(
+            user_token=recipient_token,
+            recipient_token=user_token,
+            message=f"\x01ACTION is away: {recipient_token['away_message'] or ''}\x01",
+        )
+
+    await enqueue_private_message(
+        user_token=user_token,
+        recipient_token=recipient_token,
+        message=message,
+    )
+
+
+async def handle_bot_message(
+    user_token: osuToken.Token,
+    send_to: str,
+    message: str,
+) -> Optional[int]:
+    is_channel = send_to.startswith("#")
+
+    if is_channel:
+        channel_name, client_channel_name = _get_client_channel_and_channel(
+            user_token,
+            send_to,
+        )
+
+        channel = await channelList.getChannel(channel_name)
+        if channel is None:
+            logger.warning(
+                "User attempted to send a message to an unknown channel",
+                extra={
+                    "token_id": user_token["token_id"],
+                    "username": user_token["username"],
+                    "user_id": user_token["user_id"],
+                    "channel_name": channel_name,
+                },
+            )
+
+            return ...
```

Comment:
the type annotations are not true to runtime so i don't think we should merge these -- either an error enum or None (seems the error cases aren't used for logic) would be fine for this PR